### DISCUSSION TITLE
Question on compiler generated set/get

### DISCUSSION THREAD

**Discussion by greenozon** [icsharpcode/ILSpy#3341] [General]
having the following code

```
    class Program
    {

        [CompilerGenerated]
        private string k__BackingField;

        public string DisplayText
        {
            [CompilerGenerated]
            get
            {
                return k__BackingField;
            }
            [CompilerGenerated]
            set
            {
                k__BackingField = value;
            }
        }
```

is is possible somehow to hint ILSPy to generate simple

```
        public string DisplayText { get; set; }
```
?

**Comment by siegfriedpammer** [comment]
This should happen automatically... Can you share the IL/binary?

**Reply by siegfriedpammer** [reply]
Ah I see the problem... The name of the backing field does not match the expected pattern... What compiler generated this?

**Comment by greenozon** [comment]
Very interesting! could you point me into expected pattern matching please (source code?)
the thing is that this is just an examle and in reality the code was after deobfuscator, so I'm thinking that maybe by carefull naming patterns ILSpy would then understand how to collapse backing field and create setter/getter ..etc
